src/researchhub_document/views/researchhub_post_views.py

```diff
@@ -185,3 +231,51 @@ def create_unified_doc(self, request):
             return uni_doc
         except (KeyError, TypeError) as exception:
             print('create_unified_doc: ', exception)
+
+
+def generate_doi():
+    return CROSSREF_DOI_PREFIX \
+        + ''.join(random.choice(string.ascii_lowercase + string.digits)
+                  for _ in range(CROSSREF_DOI_SUFFIX_LENGTH))
+
+
+def register_doi(created_by, title, doi, rh_post):
+    dt = datetime.today()
+    context = {
+        'timestamp': int(time.time()),
+        'first_name': created_by.author_profile.first_name,
+        'last_name': created_by.author_profile.last_name,
+        'title': title,
+        'publication_month': dt.month,
+        'publication_day': dt.day,
+        'publication_year': dt.year,
+        'doi': doi,
+        'url': f'{BASE_FRONTEND_URL}/post/{rh_post.id}/{rh_post.slug}',
+    }
+    crossref_xml = render_to_string('crossref.xml', context)
+    files = {
+        'operation': (None, 'doMDUpload'),
+        'login_id': (None, CROSSREF_LOGIN_ID),
+        'login_passwd': (None, CROSSREF_LOGIN_PASSWORD),
+        'fname': ('crossref.xml', crossref_xml),
+    }
+    crossref_response = requests.post(CROSSREF_API_URL, files=files)
+
+
+def charge_doi_fee(created_by, rh_post):
```

Comment:
We need a check to ensure the user has enough RSC to assign a DOI.

Reply:
There is a check for it before calling this function.